instrumentation/src/com/intellij/rt/coverage/instrumentation/Instrumenter.java

```diff
@@ -156,4 +159,13 @@ public boolean hasInterfaces() {
   public boolean isKotlinClass() {
     return myIsKotlinClass;
   }
+
+  private static List<MethodVisitingFilter.Builder> getMethodVisitingFilters() {
+    List<MethodVisitingFilter.Builder> result = new ArrayList<MethodVisitingFilter.Builder>();
+    ServiceLoader<MethodVisitingFilter.Builder> loader = ServiceLoader.load(MethodVisitingFilter.Builder.class);
```

Comment:
did you check how much time would be spend here for long classpathes?

Reply:
For random test from IDEA project it takes up to 180ms.
(Transformation takes 200ms)